projecttacoma/fqm-testify
# Add date paths to test resources

# Summary
Test resources now have autogenerated dates

## New Behavior
When a test resource is created, testify will parse through the dataRequirement object for that resource and add date fields for each attribute specified in the dataRequirement's dateFilters. This leads to the following several scenarios with expected outcomes
- DateFilter specifies valuePeriod and specified field accepts Period -> populate field w/ random 1-day period inside specified period
- DateFilter specifies valueDateTime and specified field accepts Period -> populate field w/ period w/ start & end equal to specified dateTime
- DateFilter specifies valueDuration and specified field accepts Period -> populate field w/ random 1-day period inside user-entered-measurementPeriod
- DateFilter specifies valuePeriod and specified field accepts dateTime -> populate field w/ random dateTime inside specified period
- DateFilter specifies valueDateTime and specified field accepts dateTime -> populate field w/ specified dateTime
- DateFilter specifies valueDuration and specified field accepts dateTime -> populate field w/ random dateTime inside user-entered-measurementPeriod
- DateFilter specifies valuePeriod and specified field accepts date -> populate field w/ random date inside specified period
- DateFilter specifies valueDateTime and specified field accepts date -> populate field w/ date associated w/ specified dateTime
- DateFilter specifies valueDuration and specified field accepts date-> populate field w/ random date inside user-entered-measurementPeriod
- DataRequirement has multiple dateFilters -> fill in fields for each filter as specified above
- DataRequirement has no dateFilters -> fill in every field on resource that can accept Period, dateTime, or date with appropriate value inside user-specified measurement period

If resource fields can accept multiple date types, prioritize Period > dateTime > date

Additionally, added a script to parse ModelInfo to produce primaryDatePath fixture that specifies all fields that can accept date info and what kind of date info it can take for each resource

## Code Changes
 - Updates to `fhir.ts` to reflect above behavior
 - Added `parseModelInfo.ts` script to create above-described fixture
 - Added test files and fixtures

# Testing Guidance
- Upload `EXM130` bundle
- Create a patient and update birthdate to `1950-01-01`
- Add an 'AnnualWellnessVisit` Encounter
- Calculate population results and see if patient calculates into denominator and IPP
- Add a Colonoscopy procedure and see if patient calculates into numerator
- Try adding all different test resources and ensure that the proper date fields are included with appropriate values

NOTE: date testing is a mess rn I have tests skipped until we figure out a way to compare dates to dateTimes
